### PR TITLE
[AMDGPU] Prevent FMINIMUM and FMAXIMUM beeing fully scalarized

### PR DESCRIPTION
**Pull request by rampitec**
This is the same logic as with FMINNUM_IEEE/FMAXNUM_IEEE.

### CODE REVIEW

File: llvm/test/CodeGen/AMDGPU/fmaximum.ll [hunk 1]
```diff
@@ -148,23 +148,35 @@ define amdgpu_ps <2 x half> @test_fmaximum_v2f16_ss(<2 x half> inreg %a, <2 x ha
...
 define amdgpu_ps <3 x half> @test_fmaximum_v3f16_vv(<3 x half> %a, <3 x half> %b) {
-; GCN-LABEL: test_fmaximum_v3f16_vv:
-; GCN:       ; %bb.0:
-; GCN-NEXT:    v_pk_maximum_f16 v0, v0, v2
-; GCN-NEXT:    v_maximum_f16 v1, v1, v3
-; GCN-NEXT:    ; return to shader part epilog
+; GFX12-SDAG-LABEL: test_fmaximum_v3f16_vv:
+; GFX12-SDAG:       ; %bb.0:
+; GFX12-SDAG-NEXT:    v_pk_maximum_f16 v0, v0, v2
+; GFX12-SDAG-NEXT:    v_pk_maximum_f16 v1, v1, v3
```
Comment (arsenm) [inline llvm/test/CodeGen/AMDGPU/fmaximum.ll:154]: We must be missing the combine to eliminate the undef high half of the vector operation too

**Review by arsenm — APPROVED** [review]
lgtm but we're not getting the undef high element to scalar operation optimization

File: llvm/test/CodeGen/AMDGPU/fmaximum.ll (hunk 1 above)
Comment (rampitec) [reply llvm/test/CodeGen/AMDGPU/fmaximum.ll:154]: Yes, although that is a separate optimization unrelated to a specific opcode. I think what we need is a general optimization checks if a v2 operation has unused high element and if a same scalar operation is legal then splits it.

Comment (arsenm) [reply llvm/test/CodeGen/AMDGPU/fmaximum.ll:154]: I think we already have this but I don't immediately see where it is

**Pull request merged**